flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/DefaultVertexParallelismStore.java

```
@@ -19,13 +19,50 @@
 package org.apache.flink.runtime.scheduler;
 
 import org.apache.flink.runtime.jobgraph.JobVertexID;
+import org.apache.flink.runtime.jobmaster.JobResourceRequirements;
+import org.apache.flink.runtime.jobmaster.JobVertexResourceRequirements;
 
 import java.util.HashMap;
 import java.util.Map;
 import java.util.Optional;
 
 /** Maintains the configured parallelisms for vertices, which should be defined by a scheduler. */
 public class DefaultVertexParallelismStore implements MutableVertexParallelismStore {
+
+    /**
+     * Create a new {@link VertexParallelismStore} that reflects given {@link
+     * JobResourceRequirements}.
+     *
+     * @param oldVertexParallelismStore old vertex parallelism store that serves as a base for the
+     *     new one
+     * @param jobResourceRequirements to apply over the old vertex parallelism store
+     * @return new vertex parallelism store iff it was updated
+     */
+    public static Optional<VertexParallelismStore> applyJobResourceRequirements(
+            VertexParallelismStore oldVertexParallelismStore,
+            JobResourceRequirements jobResourceRequirements) {
+        final DefaultVertexParallelismStore newVertexParallelismStore =
+                new DefaultVertexParallelismStore();
+        boolean changed = false;
+        for (final JobVertexID jobVertexId : jobResourceRequirements.getJobVertices()) {
+            final VertexParallelismInformation oldVertexParallelismInfo =
+                    oldVertexParallelismStore.getParallelismInfo(jobVertexId);
+            final int parallelism =
+                    jobResourceRequirements
+                            .findParallelism(jobVertexId)
```

I thought a put has to be complete and cover all vertices. Then find should either always find something, or fail because the request is invalid.